src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/task.py

```diff
@@ -119,8 +126,8 @@ def acr_task_create(cmd,  # pylint: disable=too-many-locals
                         scope='repo'
                     )
                 ),
-                source_trigger_events=[SourceTriggerEvent.commit.value],
-                status=TriggerStatus.enabled.value if commit_trigger_enabled else TriggerStatus.disabled.value,
+                source_trigger_events=source_trigger_events,
+                status=TriggerStatus.enabled.value if source_trigger_events else TriggerStatus.disabled.value,
```

Comment:
Do you need to check `if source_trigger_events` again?

Also it seems redundant for the client to provide this `status`, can the server side determine and return the correct status based on the value of `source_trigger_events`? Or is it possible to have a case where `source_trigger_events` is not empty but `status` is disabled? Or `source_trigger_events` is empty but `status` is enabled?

Reply:
See fix in most recent update.